File: lib/netra/core/entities/bookmark.dart
class Bookmark {
  const Bookmark({
    required this.id,
    required this.url,
    required this.title,
    required this.createdAt,
    this.folderId,
    this.faviconPath,
    this.sortOrder = 0,
  });

  final String id;
  final String url;
  final String title;
  final String? folderId;
  final String? faviconPath;
  final DateTime createdAt;
  final int sortOrder;
}

File: rust/core/entities/bookmark.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Bookmark {
    pub id: String,
    pub title: String,
    pub url: String,
    pub folder: Option<String>,
    pub created_at: i64,
}

impl Bookmark {
    pub fn new(id: String, title: String, url: String) -> Self {
        Self {
            id,
            title,
            url,
            folder: None,
            created_at: 0,
        }
    }
}
